SettingsWatchdog/logging.hpp

```diff
@@ -1,30 +1,60 @@
 #pragma once
 DISABLE_ANALYSIS
+#include <format>
 #include <iosfwd>
 #include <string>
 #include <vector>
 
 #include <boost/any.hpp>
-#include <boost/log/sources/global_logger_storage.hpp>
-#include <boost/log/trivial.hpp>
+#include <boost/nowide/convert.hpp>
+#include <plog/Log.h>
 REENABLE_ANALYSIS
 
+class ScopeMarker
+{
+public:
+    ScopeMarker(char const* name);
+    ~ScopeMarker();
+};
+
+#define BOOST_LOG_FUNC()            \
+    ScopeMarker const CuRrEnT_ScOpE \
+    {                               \
+        __FUNCTION__                \
+    }
+
 enum class severity_level
 {
-    trace,
-    debug,
-    info,
-    warning,
-    error,
-    fatal,
+    trace = plog::verbose,
+    debug = plog::debug,
+    info = plog::info,
+    warning = plog::warning,
+    error = plog::error,
+    fatal = plog::fatal,
```

Comment:
We shouldn't really need this anymore. Instead, change `WDLOG` to accept `plog::Severity` values instead, and then adjust the other functions below to work on that type as well. Then the only change in the rest of the code would be to change `WDLOG(trace)` to be `WDLOG(verbose)` instead, which isn't that big a deal.

Reply:
I added an alias for `trace = verbose` and barely needed to change anything.